core/chain/node.go

```diff
@@ -0,0 +1,69 @@
+package chain
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/AudiusProject/audius-protocol/core/common"
+	"github.com/AudiusProject/audius-protocol/core/config"
+	cfg "github.com/cometbft/cometbft/config"
+	nm "github.com/cometbft/cometbft/node"
+	"github.com/cometbft/cometbft/p2p"
+	"github.com/cometbft/cometbft/privval"
+	"github.com/cometbft/cometbft/proxy"
+	"github.com/jackc/pgx/v5/pgxpool"
+)
+
+func NewNode(logger *common.Logger, c *config.Config, pool *pgxpool.Pool) (*nm.Node, error) {
+	homeDir := c.HomeDir
+
+	config := cfg.DefaultConfig()
+	config.SetRoot(homeDir)
+
+	// postgres indexer config
+	config.TxIndex.Indexer = "psql"
+	config.TxIndex.PsqlConn = c.PSQLConn
+	config.TxIndex.TableBlocks = "core_blocks"
+	config.TxIndex.TableTxResults = "core_tx_results"
+	config.TxIndex.TableEvents = "core_events"
+	config.TxIndex.TableAttributes = "core_attributes"
```

Comment:
Since we already have a `blocks` table, we needed the ability to rename these. We're using a forked version until cometbft supports custom table names for their postgres indexer.